ophyd/status.py

```diff
@@ -203,11 +208,32 @@ def finished_cb(self):
                                  "property instead.")
 
     @_locked
-    def add_callback(self, cb):
+    def add_callback(self, callback):
+        """
+        Register a callback to be called once when the Status finishes.
+
+        The callback will be called exactly once. If the Status is finished
+        before a callback is added, it will be called immediately. This is
+        threadsafe.
+
+        The callback will be called regardless of success of failure. The
+        callback has access to this status object, so it can distinguish success
+        or failure by inspecting the object.
+
+        Parameters
+        ----------
+        callback: callable
+            Expected signature: ``callback(status)``.
+
+            The signature ``callback()`` is also supported for
+            backward-compatibility but will issue warnings. Support will be
+            removed in a future release of ophyd.
+        """
+        # Handle func with signature callback() for back-compat.
+        callback = adapt_old_callback_signature(callback, self.__wrapper_weakrefs)
         if self.done:
-            cb()
-        else:
-            self._callbacks.append(cb)
+            callback()
```

Comment:
sholudn't we be passing the status objec in here?

Reply:
Yep, fixed.